### PR TITLE
Rename entities

### LINKED ISSUE #16: Allow Entity Editor to Rename Entities

### PR DESCRIPTION
**Pull request by tlein**
Resolves #16

### CODE REVIEW

File: src/duckling/entityeditor/entity-editor.component.ts [hunk 1]
```diff
@@ -6,26 +6,50 @@ import {Entity, EntityKey, EntitySystemService, AttributeDefaultService, Attribu
...
         <div *ngIf="selection?.selectedEntity">
-            <span>
-                Entity {{selection.selectedEntity}}
-                <button
-                    [disableRipple]=true
-                    md-button title="Delete Entity"
-                    (click)="deleteEntity()">
-                    Delete
-                </button>
+            <span class="entity-name-row">
```
Comment (jeffswenson) [inline src/duckling/entityeditor/entity-editor.component.ts]: You should extract this into it's own component.

File: src/duckling/entitysystem/entity-system.service.ts [hunk 2]
```diff
@@ -72,6 +72,16 @@ export class EntitySystemService {
     }
 
     /**
+     * Renames an entity
+     * @param  oldEntityKey The old key of the entity to be renamed
+     * @param  newEntityKey The new key of the entity to be renamed
+     * @param  mergeKey     Used to merge updates.
+     */
+    renameEntity(oldEntityKey : EntityKey, newEntityKey : EntityKey, mergeKey? : any) {
```
Comment (jeffswenson) [inline src/duckling/entitysystem/entity-system.service.ts:80]: You should write a unit test for this.

Comment (tlein) [reply src/duckling/entitysystem/entity-system.service.ts:80]: Should we unit test the entirety of the entity-system.service? We don't have any yet

File: src/duckling/entityeditor/entity-name.component.ts [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+            <dk-delete-button (click)="onDeleteClicked()"></dk-delete-button>
+        </span>
+    `
+})
+export class EntityNameComponent implements OnChanges {
+    @Input() currentSelectedEntity : EntityKey;
+    @Output() deleteEntity = new EventEmitter<any>();
+    @Output() renameEntity = new EventEmitter<any>();
+
+    entityName : string;
```
Comment (jeffswenson) [inline src/duckling/entityeditor/entity-name.component.ts]: I think you can get rid of entityName (or beginEditName) and validNewName.

**Pull request merged**